readthedocs/api/v3/serializers.py

```diff
@@ -322,13 +323,15 @@ class VersionURLsSerializer(BaseLinksSerializer, serializers.Serializer):
     dashboard = VersionDashboardURLsSerializer(source="*")
 
     def get_documentation(self, obj):
-        return Resolver().resolve_version(
+        resolver = getattr(self.parent, "resolver", Resolver())
```

Comment:
Can we pass this to the constructor? That way is more explicit, rather than relying on the attribute being present in the parent serialzer.

Reply:
Looks like you could pass this in the context of the serializer, something like:

```python
class ChildSerializer:
    def get_documentation(self, obj):
        resolver = self.context.get('resolver')
        ...

class ParentSerializer:
    child = ChildSerializer()
    def __init__(self, resolver):
        self.fields['child'].context['resolver'] = resolver
```

Reply:
Thanks. It sounds similar to what I did but more standardized. I found some documentation about this at https://www.django-rest-framework.org/api-guide/serializers/#including-extra-context

I opened https://github.com/readthedocs/readthedocs.org/issues/11280 to track this refactor as it's not a blocker for now and it may not be required in the future anyways.